### PR TITLE
add io.itch.nordup.TheGates

### PR DESCRIPTION
**Pull request by CredibleOpossum**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
I'm a third-party with permission, not on any public medium, I'm sure https://github.com/Nordup/ would be willing to give me thumbs on up this submission for proof.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
closed-source application, I am working with the developer so any high-importance changes could be worked on. 
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: io.itch.nordup.TheGates.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - type: file
+        path: icons/icon_32.png
+      - type: file
+        path: icons/icon_64.png
+      - type: file
+        path: icons/icon_128.png
+      - type: file
+        path: icons/icon_256.png
+      - type: file
+        path: icons/icon_512.png
```
Comment (barthalion) [inline io.itch.nordup.TheGates.yml:55]: Unless also used internally, 512x512 is sufficient.

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "automerge-flathubbot-prs": true,
```
Comment (barthalion) [inline flathub.json:2]: This line doesn't make any sense unless you add x-data-checker stanza to automate updates.

File: io.itch.nordup.TheGates.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+finish-args:
```
Comment (barthalion) [inline io.itch.nordup.TheGates.yml:56]: Typically, `finish-args` are above `modules`.

**Comment by barthalion** [comment]
bot, build io.itch.nordup.TheGates

**Comment by barthalion** [comment]
The manifest uses a mix of tabs and spaces and is not a valid YAML.

File: flathub.json (hunk 2 above)
Comment (CredibleOpossum) [reply flathub.json]: My bad, honestly wasn't too aware of it, just saw other projects use it

**Comment by CredibleOpossum** [comment]
bot, build io.itch.nordup.TheGates

**Comment by CredibleOpossum** [comment]
The screenshots seem to load fine when built locally from my flathub repo, but not when installed from flathubbot, is this a misconfiguration on my part?

File: io.itch.nordup.TheGates.yml (hunk 1 above)
Comment (CredibleOpossum) [reply io.itch.nordup.TheGates.yml]: fixed!

File: io.itch.nordup.TheGates.yml (hunk 3 above)
Comment (CredibleOpossum) [reply io.itch.nordup.TheGates.yml]: fixed!

**Comment by CredibleOpossum** [comment]
bot, build io.itch.nordup.TheGates

File: gates.desktop [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Categories=Game;
+Exec=
```
Comment (hfiguiere) [inline gates.desktop:3]: exec should be set here too.

File: io.itch.nordup.TheGates.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+app-id: io.itch.nordup.TheGates
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: TheGates
+tags:
+  - proprietary
+cleanup:
+  - '*.cmake'
```
Comment (hfiguiere) [inline io.itch.nordup.TheGates.yml:9]: I'm pretty there is more to clean like `/include` and maybe `*.a`

Comment (CredibleOpossum) [reply io.itch.nordup.TheGates.yml:9]: I've also found a debug directory that serves no purpose in the build, it says it was deleted when building but the build directory still has it, does this mean it's not actually being deleted?

File: io.itch.nordup.TheGates.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  - '*.a'
+  - 'debug'
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.itch.nordup.TheGates.yml:11]:
```suggestion
```

[the remark this answers is not in the document]
Comment (CredibleOpossum) [reply io.itch.nordup.TheGates.yml:11]: oh, whoops

File: io.itch.nordup.TheGates.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+tags:
+  - proprietary
+cleanup:
+  - '*.cmake'
+  - '*.a'
+  - '/include'
+finish-args:
+  - '--share=network'
+  - '--share=ipc'
+  - '--socket=wayland'
```
Comment (barthalion) [inline io.itch.nordup.TheGates.yml:15]:
```suggestion
```

File: io.itch.nordup.TheGates.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  - '--socket=pulseaudio'
+  - '--socket=fallback-x11'
```
Comment (barthalion) [inline io.itch.nordup.TheGates.yml:17]:
```suggestion
  - '--socket=x11'
```

**Review by barthalion — CHANGES_REQUESTED** [review]
It doesn't actually work on Wayland.

**Comment by CredibleOpossum** [comment]
> It doesn't actually work on Wayland.

huh, I assumed Godot did, that's too bad.

**Comment by CredibleOpossum** [comment]
bot, build io.itch.nordup.TheGates

**Comment by CredibleOpossum** [comment]
for further reference, is there an easy way to diagnose this without having access to wayland? My drivers aren't the best at the moment.

**Comment by barthalion** [comment]
I don't think there is, you can either test locally (if it works) or in a VM, but that's about it.

/merge

**Pull request closed without merging**